main.py

```diff
@@ -18,29 +18,49 @@ def main(cfg: DictConfig):
         # Convert config to dict type
         config=OmegaConf.to_container(cfg, resolve=True),
         mode="disabled" if cfg.disable_wandb else "online",
-        tags=cfg.get("tags", None),
+        tags=[cfg.basename if cfg.run_baseline else cfg.unadaptname], # when retrieving baseline result, we search for the latest run with the basename tag. (it might be better to search with run id to ensure uniqueness) 
         save_code=True,
+        name=cfg.basename if cfg.run_baseline else cfg.unadaptname,
```

Comment:
Why the choice to specify a name? If you want it, you should probably include a random identifier along with it so we don't just end up with a bunch of runs all called e.g. `Pruning`